crates/rattler/src/install/link_script.rs

```diff
@@ -0,0 +1,251 @@
+                }
+            }
+        } else {
+            match self {
+                LinkScriptType::PreUnlink => {
+                    format!("bin/.{name}-pre-unlink.sh")
+                }
+                LinkScriptType::PostLink => {
+                    format!("bin/.{name}-post-link.sh")
+                }
+            }
+        }
+    }
+}
+
+impl ToString for LinkScriptType {
+    fn to_string(&self) -> String {
+        match self {
+            LinkScriptType::PreUnlink => "pre-unlink".to_string(),
+            LinkScriptType::PostLink => "post-link".to_string(),
+        }
+    }
+}
+
+/// Run the link scripts for a given package
+pub fn run_link_scripts<'a>(
+    link_script_type: LinkScriptType,
+    prefix_records: impl Iterator<Item = &'a PrefixRecord>,
+    target_prefix: &Path,
+    platform: &Platform,
+) -> Result<(), LinkScriptError> {
+    let mut env = HashMap::new();
+    env.insert(
+        "PREFIX".to_string(),
+        target_prefix.to_string_lossy().to_string(),
+    );
+
+    // prefix records are topologically sorted, so we can be sure that all dependencies are
+    // installed before the package itself.
+    for record in prefix_records {
+        let prec = &record.repodata_record.package_record;
+        let link_file = target_prefix.join(&link_script_type.get_path(prec, platform));
+
+        if link_file.exists() {
+            env.insert(
+                "PKG_NAME".to_string(),
+                prec.name.as_normalized().to_string(),
+            );
+            env.insert("PKG_VERSION".to_string(), prec.version.to_string());
+            env.insert("PKG_BUILDNUM".to_string(), prec.build_number.to_string());
+
+            let shell = if platform.is_windows() {
+                ShellEnum::CmdExe(CmdExe)
+            } else {
+                ShellEnum::Bash(Bash)
+            };
+
+            tracing::info!(
+                "Running {} script for {}",
+                link_script_type.to_string(),
+                prec.name.as_normalized()
+            );
+
+            match rattler_shell::run_in_environment(
+                target_prefix,
+                &[&link_file.to_string_lossy()],
+                shell,
+                &env,
+            ) {
+                Ok(o) if o.status.success() => {}
+                Ok(o) => {
+                    tracing::warn!("Error running post-link script. Status: {:?}", o.status);
+                    tracing::warn!("  stdout: {}", String::from_utf8_lossy(&o.stdout));
+                    tracing::warn!("  stderr: {}", String::from_utf8_lossy(&o.stderr));
+                }
+                Err(e) => {
+                    tracing::error!("Error running post-link script: {:?}", e);
+                }
+            }
+
+            let message_file = target_prefix.join(".messages.txt");
+            if message_file.exists() {
+                let message = std::fs::read_to_string(&message_file)?;
+                tracing::info!(
```

Comment:
Would it make sense to return these, or store them in the driver, so a user can decide what to do with them?